### PR TITLE
Transformations: Preserve base threshold when using Config from query results

### LINKED ISSUE #90201: Dashboard: Thresholds configured by query overwrite other customizations
### What happened?

I've set up some customizations on gauge threshold (levels, colors) and wanted to use the Transformation "config by query result" to set an *additional* customization, a dynamic threshold.
This dynamic threshold works, but all other costumizations are overwritten. For instance, the base threshold color was set by me to yellow, but with the transformation, again the default green color is applied. All other thresholds are not applied.

When multiple thresholds are configured by the Transformation, only the last is applied.

### What did you expect to happen?

The dynamic threshold and its color are applied additionally to the other settings.
All dynamic threshold are applied.

### Did this work before?

Did not try before.

### How do we reproduce it?

1. set up some threshold customizations on a gauge in the default settings pane.
2. add the transformation "config by query result" for a threshold.

### Is the bug inside a dashboard panel?

| Key | Value |
|--|--|
| Panel | gauge @ 11.0.0 |
| Grafana | Grafana v11.0.0 (277ef258d4) // Open Source |
<details><summary>Panel debug snapshot dashboard</summary>

```json
{
  "panels": [
    {
      "datasource": {
        "type": "grafana",
        "uid": "grafana"
      },
      "fieldConfig": {
        "defaults": {
          "custom": {
            "neutral": 0
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "dark-purple",
                "value": null
              },
              {
                "color": "blue",
                "value": 0
              }
            ]
          },
          "color": {
            "mode": "thresholds"
          },
          "fieldMinMax": true,
          "max": 15000,
          "min": -1000,
          "unit": "watt"
        },
        "overrides": [
          {
            "matcher": {
              "id": "byName",
              "options": "PV AC Power"
            },
            "properties": [
              {
                "id": "max",
                "value": 1500
              },
              {
                "id": "min",
                "value": 0
              }
            ]
          }
        ]
      },
      "gridPos": {
        "h": 13,
        "w": 15,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "reduceOptions": {
          "values": false,
          "calcs": [
            "lastNotNull"
          ],
          "fields": ""
        },
        "orientation": "auto",
        "showThresholdLabels": false,
        "showThresholdMarkers": true,
        "sizing": "auto",
        "minVizWidth": 75,
        "minVizHeight": 75,
        "text": {}
      },
      "pluginVersion": "11.0.0",
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "type": "grafana",
            "uid": "grafana"
          },
          "queryType": "snapshot",
          "snapshot": [
            {
              "schema": {
                "refId": "A",
                "meta": {
                  "typeVersion": [
                    0,
                    0
                  ],
                  "preferredVisualisationType": "graph",
                  "executedQueryString": "SELECT last(\"value\") FROM \"W\" WHERE \"entity_id\"::tag = '16_7_0' AND time >= 1720438861103ms and time <= 1720460461103ms ORDER BY time ASC"
                },
                "name": "Grid Power",
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {}
                  },
                  {
                    "name": "Grid Power",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64",
                      "nullable": true
                    },
                    "config": {
                      "displayNameFromDS": "Grid Power"
                    }
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1720460453222
                  ],
                  [
                    541
                  ]
                ]
              }
            },
            {
              "schema": {
                "refId": "B",
                "meta": {
                  "typeVersion": [
                    0,
                    0
                  ],
                  "preferredVisualisationType": "graph",
                  "executedQueryString": "SELECT last(\"value\") FROM \"W\" WHERE \"entity_id\"::tag = 'pv_dachbaga_power' AND time >= 1720438861103ms and time <= 1720460461103ms ORDER BY time ASC"
                },
                "name": "PV AC Power",
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {}
                  },
                  {
                    "name": "PV AC Power",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64",
                      "nullable": true
                    },
                    "config": {
                      "displayNameFromDS": "PV AC Power"
                    }
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1720460449869
                  ],
                  [
                    53
                  ]
                ]
              }
            },
            {
              "schema": {
                "refId": "C",
                "meta": {
                  "typeVersion": [
                    0,
                    0
                  ],
                  "preferredVisualisationType": "graph",
                  "executedQueryString": "SELECT last(\"value\") FROM \"W\" WHERE \"entity_id\"::tag = 'pv_dachbaga_limit_nonpersistent_absolute' AND time >= 1720438861103ms and time <= 1720460461103ms ORDER BY time ASC"
                },
                "name": "PV Limit",
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {}
                  },
                  {
                    "name": "PV Limit",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64",
                      "nullable": true
                    },
                    "config": {
                      "displayNameFromDS": "PV Limit"
                    }
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1720448828195
                  ],
                  [
                    795
                  ]
                ]
              }
            },
            {
              "schema": {
                "refId": "D",
                "meta": {
                  "type": "timeseries-multi",
                  "typeVersion": [
                    0,
                    1
                  ]
                },
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {}
                  },
                  {
                    "name": "D",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64",
                      "nullable": true
                    },
                    "config": {}
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1720448828195
                  ],
                  [
                    0
                  ]
                ]
              }
            }
          ]
        }
      ],
      "title": "Reproduced with embedded data",
      "transformations": [
        {
          "id": "configFromData",
          "options": {
            "applyTo": {
              "id": "byName",
              "options": "PV AC Power"
            },
            "configRefId": "D",
            "mappings": [
              {
                "fieldName": "D",
                "handlerArguments": {
                  "threshold": {
                    "color": "yellow"
                  }
                },
                "handlerKey": "threshold1"
              }
            ]
          }
        },
        {
          "id": "configFromData",
          "options": {
            "applyTo": {
              "id": "byName",
              "options": "PV AC Power"
            },
            "configRefId": "C",
            "mappings": [
              {
                "fieldName": "PV Limit",
                "handlerArguments": {
                  "threshold": {
                    "color": "orange"
                  }
                },
                "handlerKey": "threshold1"
              }
            ]
          }
        }
      ],
      "transparent": true,
      "type": "gauge"
    },
    {
      "gridPos": {
        "h": 7,
        "w": 9,
        "x": 15,
        "y": 0
      },
      "id": 5,
      "options": {
        "content": "<table width=\"100%\">\n    <tr>\n      <th width=\"2%\">Panel</th>\n      <td >gauge @ 11.0.0</td>\n    </tr>\n    <tr>\n      <th>Queries</th>\n      <td>A[influxdb], B[influxdb], C[influxdb], D[__expr__]</td>\n    </tr>\n    <tr>\n      <th>Transform</th>\n      <td>configFromData, configFromData</td>\n  </tr>\n    <tr><th>Data</th><td> 4 frames, 8 fields, 4 rows </td></tr>\n    \n    <tr>\n      <th>Grafana</th>\n      <td>Grafana v11.0.0 (277ef258d4) // Open Source</td>\n    </tr>\n  </table>",
        "mode": "html"
      },
      "title": "Debug info",
      "type": "text"
    },
    {
      "id": 6,
      "title": "Original Panel JSON",
      "type": "text",
      "gridPos": {
        "h": 13,
        "w": 9,
        "x": 15,
        "y": 7
      },
      "options": {
        "content": "{\n  \"datasource\": {\n    \"type\": \"influxdb\",\n    \"uid\": \"d9689cd3-6a86-4567-a0f9-cb621f02ff69\"\n  },\n  \"fieldConfig\": {\n    \"defaults\": {\n      \"custom\": {\n        \"neutral\": 0\n      },\n      \"mappings\": [],\n      \"thresholds\": {\n        \"mode\": \"absolute\",\n        \"steps\": [\n          {\n            \"color\": \"dark-purple\",\n            \"value\": null\n          },\n          {\n            \"color\": \"blue\",\n            \"value\": 0\n          }\n        ]\n      },\n      \"color\": {\n        \"mode\": \"thresholds\"\n      },\n      \"fieldMinMax\": true,\n      \"max\": 15000,\n      \"min\": -1000,\n      \"unit\": \"watt\"\n    },\n    \"overrides\": [\n      {\n        \"matcher\": {\n          \"id\": \"byName\",\n          \"options\": \"PV AC Power\"\n        },\n        \"properties\": [\n          {\n            \"id\": \"max\",\n            \"value\": 1500\n          },\n          {\n            \"id\": \"min\",\n            \"value\": 0\n          }\n        ]\n      }\n    ]\n  },\n  \"gridPos\": {\n    \"h\": 5,\n    \"w\": 6,\n    \"x\": 0,\n    \"y\": 0\n  },\n  \"id\": 14,\n  \"options\": {\n    \"reduceOptions\": {\n      \"values\": false,\n      \"calcs\": [\n        \"lastNotNull\"\n      ],\n      \"fields\": \"\"\n    },\n    \"orientation\": \"auto\",\n    \"showThresholdLabels\": false,\n    \"showThresholdMarkers\": true,\n    \"sizing\": \"auto\",\n    \"minVizWidth\": 75,\n    \"minVizHeight\": 75,\n    \"text\": {}\n  },\n  \"pluginVersion\": \"11.0.0\",\n  \"targets\": [\n    {\n      \"alias\": \"Grid Power\",\n      \"datasource\": {\n        \"type\": \"influxdb\",\n        \"uid\": \"d9689cd3-6a86-4567-a0f9-cb621f02ff69\"\n      },\n      \"groupBy\": [],\n      \"measurement\": \"W\",\n      \"orderByTime\": \"ASC\",\n      \"policy\": \"default\",\n      \"refId\": \"A\",\n      \"resultFormat\": \"time_series\",\n      \"select\": [\n        [\n          {\n            \"params\": [\n              \"value\"\n            ],\n            \"type\": \"field\"\n          },\n          {\n            \"params\": [],\n            \"type\": \"last\"\n          }\n        ]\n      ],\n      \"tags\": [\n        {\n          \"key\": \"entity_id::tag\",\n          \"operator\": \"=\",\n          \"value\": \"16_7_0\"\n        }\n      ]\n    },\n    {\n      \"alias\": \"PV AC Power\",\n      \"datasource\": {\n        \"type\": \"influxdb\",\n        \"uid\": \"d9689cd3-6a86-4567-a0f9-cb621f02ff69\"\n      },\n      \"groupBy\": [],\n      \"hide\": false,\n      \"measurement\": \"W\",\n      \"orderByTime\": \"ASC\",\n      \"policy\": \"default\",\n      \"refId\": \"B\",\n      \"resultFormat\": \"time_series\",\n      \"select\": [\n        [\n          {\n            \"params\": [\n              \"value\"\n            ],\n            \"type\": \"field\"\n          },\n          {\n            \"params\": [],\n            \"type\": \"last\"\n          }\n        ]\n      ],\n      \"tags\": [\n        {\n          \"key\": \"entity_id::tag\",\n          \"operator\": \"=\",\n          \"value\": \"pv_dachbaga_power\"\n        }\n      ]\n    },\n    {\n      \"alias\": \"PV Limit\",\n      \"datasource\": {\n        \"type\": \"influxdb\",\n        \"uid\": \"d9689cd3-6a86-4567-a0f9-cb621f02ff69\"\n      },\n      \"groupBy\": [],\n      \"hide\": false,\n      \"measurement\": \"W\",\n      \"orderByTime\": \"ASC\",\n      \"policy\": \"default\",\n      \"refId\": \"C\",\n      \"resultFormat\": \"time_series\",\n      \"select\": [\n        [\n          {\n            \"params\": [\n              \"value\"\n            ],\n            \"type\": \"field\"\n          },\n          {\n            \"params\": [],\n            \"type\": \"last\"\n          }\n        ]\n      ],\n      \"tags\": [\n        {\n          \"key\": \"entity_id::tag\",\n          \"operator\": \"=\",\n          \"value\": \"pv_dachbaga_limit_nonpersistent_absolute\"\n        }\n      ]\n    },\n    {\n      \"datasource\": {\n        \"name\": \"Expression\",\n        \"type\": \"__expr__\",\n        \"uid\": \"__expr__\"\n      },\n      \"expression\": \"$C*0\",\n      \"hide\": false,\n      \"refId\": \"D\",\n      \"type\": \"math\"\n    }\n  ],\n  \"title\": \"Current Power\",\n  \"transformations\": [\n    {\n      \"id\": \"configFromData\",\n      \"options\": {\n        \"applyTo\": {\n          \"id\": \"byName\",\n          \"options\": \"PV AC Power\"\n        },\n        \"configRefId\": \"D\",\n        \"mappings\": [\n          {\n            \"fieldName\": \"D\",\n            \"handlerArguments\": {\n              \"threshold\": {\n                \"color\": \"yellow\"\n              }\n            },\n            \"handlerKey\": \"threshold1\"\n          }\n        ]\n      }\n    },\n    {\n      \"id\": \"configFromData\",\n      \"options\": {\n        \"applyTo\": {\n          \"id\": \"byName\",\n          \"options\": \"PV AC Power\"\n        },\n        \"configRefId\": \"C\",\n        \"mappings\": [\n          {\n            \"fieldName\": \"PV Limit\",\n            \"handlerArguments\": {\n              \"threshold\": {\n                \"color\": \"orange\"\n              }\n            },\n            \"handlerKey\": \"threshold1\"\n          }\n        ]\n      }\n    }\n  ],\n  \"transparent\": true,\n  \"type\": \"gauge\"\n}",
        "mode": "code",
        "code": {
          "language": "json",
          "showLineNumbers": true,
          "showMiniMap": true
        }
      }
    },
    {
      "id": 3,
      "title": "Data from panel above (after transformations)",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 15,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": true,
          "refId": "A"
        }
      ]
    },
    {
      "id": 100,
      "title": "Data (before transformations)",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 24,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": false,
          "refId": "A"
        }
      ]
    }
  ],
  "schemaVersion": 37,
  "title": "Debug: Current Power // 2024-07-08 19:41:07",
  "tags": [
    "debug",
    "debug-gauge"
  ],
  "time": {
    "from": "2024-07-08T11:41:07.681Z",
    "to": "2024-07-08T17:41:07.682Z"
  }
}
```
</details>

### Environment (with versions)?

Grafana: Grafana v11.0.0 (277ef258d4)
OS: Home Assistant/Docker
Browser: Firefox 127.0.2

### Grafana platform?

Docker

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by gelicia**
**What is this feature?**

In Grafana, there are places to add in thresholds outside the thresholds panel option. In the panel option, it is required to have a default "base" threshold, that is -Infinity to whatever the first threshold is. For places like the Config from Query Results transformation, you can set a threshold from an existing query, but it ignores the panel option base threshold and hardcodes in green as the base color ([src](https://github.com/grafana/grafana/blob/main/public/app/features/transformers/fieldToConfigMapping/fieldToConfigMapping.ts#L172)).

This removes that hardcoded base and uses the panel config base threshold instead.

http://localhost:3000/d/Juj4_7ink/transforms-config-from-query?orgId=1

before (blue Base threshold overriden to default green by transform):
image

after (blue Base is preserved):
image

**Why do we need this feature?**

Green is not an acceptable base threshold for every use case, and should be configurable.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/90201
Related to https://github.com/grafana/support-escalations/issues/13536

**Special notes for your reviewer:**

<details><summary>Prometheus example provided in #91870</summary>

 Create two prometheus queries. Name one "query" and have it be `vector(99)`, have the other be "threshold" and have it be `vector(100)`. The "threshold" query result will be the threshold for the "query" data - so 100 will be the threshold. 

Create a "config from query results" transformation. Config query is threshold, apply to fields with type, apply to options Number (2). Ignore the time field. `vector(100)` use as threshold, last *, and choose a neutral color like blue.

On the panel options, choose red as a base threshold. Green is hardcoded, as long as we are not seeing green, one of the configured thresholds is being used. 

Play with the `query` value. 99 should be the set base color from the panel options. 101 will be the set threshold color from the transformation. If you see green anywhere, it did not work. 
</details>

<details><summary>Test Data Dashboard</summary>

This dashboard uses the last random walk value to be the value, and the threshold comes from a static CSV value of 50. Everything over 50 will the transformation threshold color of blue, and everything under 50 will be the base threshold color of red.

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 231,
  "links": [],
  "panels": [
    {
      "datasource": {
        "type": "grafana-testdata-datasource",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "red",
                "value": null
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 1,
      "options": {
        "colorMode": "value",
        "graphMode": "area",
        "justifyMode": "auto",
        "orientation": "auto",
        "percentChangeColorMode": "standard",
        "reduceOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "fields": "",
          "values": false
        },
        "showPercentChange": false,
        "textMode": "auto",
        "wideLayout": true
      },
      "pluginVersion": "11.4.0-pre",
      "targets": [
        {
          "datasource": {
            "type": "grafana-testdata-datasource",
            "uid": "PD8C576611E62080A"
          },
          "refId": "query",
          "scenarioId": "random_walk"
        },
        {
          "csvContent": "value\n50",
          "datasource": {
            "type": "grafana-testdata-datasource",
            "uid": "PD8C576611E62080A"
          },
          "refId": "threshold",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "transformations": [
        {
          "id": "configFromData",
          "options": {
            "configRefId": "threshold",
            "mappings": [
              {
                "fieldName": "value",
                "handlerArguments": {
                  "threshold": {
                    "color": "blue"
                  }
                },
                "handlerKey": "threshold1"
              }
            ]
          }
        }
      ],
      "type": "stat"
    }
  ],
  "preload": false,
  "schemaVersion": 40,
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "2024-11-20T09:57:29.457Z",
    "to": "2024-11-20T21:57:29.457Z"
  },
  "timepicker": {},
  "timezone": "browser",
  "title": "Threshold example",
  "uid": "ae4isxh9vxvcwa",
  "version": 2,
  "weekStart": ""
}
```
</details>

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 1]
```diff
@@ -347,6 +347,18 @@ export function setFieldConfigDefaults(config: FieldConfig, defaults: FieldConfi
...
+  // if we have a base threshold set by default but not on the config, we need to merge it in
+  const defaultBaseStep =
+    defaults.thresholds?.mode === ThresholdsMode.Absolute &&
+    defaults.thresholds?.steps.find((step) => step.value === -Infinity);
+  if (
+    config.thresholds?.mode === ThresholdsMode.Absolute &&
+    !config.thresholds.steps.some((step) => step.value === -Infinity) &&
+    defaultBaseStep
+  ) {
+    config.thresholds.steps = [defaultBaseStep, ...config.thresholds.steps];
```
Comment (gelicia) [inline packages/grafana-data/src/field/fieldOverrides.ts:360]: order matters here unforch

**Review by leeoniya — APPROVED** [review]
404: Bugs Not Found

but can you take a peek @dprokop :pray:

**Review by dprokop — APPROVED** [review]
Think this looks goo! 

But it feels like this is introducing some sort of inconsistency in the transform behavior. It kind of merges panel and transformation provided config, but not really as it only applies to the base. Otherwise it will replace the panel config, as it's literally a "config from DS"? 

I think it's fine, but maybe worth mentioning somewhere in the docs that the base threshold will be preserved when this config is applied.

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 2]
```diff
@@ -347,6 +347,18 @@ export function setFieldConfigDefaults(config: FieldConfig, defaults: FieldConfi
...
+
+  // if we have a base threshold set by default but not on the config, we need to merge it in
+  const defaultBaseStep =
+    defaults?.thresholds?.mode === ThresholdsMode.Absolute &&
+    defaults.thresholds?.steps.find((step) => step.value === -Infinity);
+  if (
+    config.thresholds?.mode === ThresholdsMode.Absolute &&
+    !config.thresholds.steps.some((step) => step.value === -Infinity) &&
+    defaultBaseStep
+  ) {
```
[the remark this answers is not in the document]
Comment (gelicia) [reply packages/grafana-data/src/field/fieldOverrides.ts:359]: For the transformation anyway, the config you set is only by absolute value - it made sense to me to require that to be consistent with the panel setting to work. It wouldn't be impossible that someone would want the base threshold by percentage and the config by absolute value but I'd rather require them both to be the same for this PR.

**Comment by gelicia** [comment]
> maybe worth mentioning somewhere in the docs that the base threshold will be preserved when this config is applied.

@dprokop trying to keep this both short and informative. What do you think? 

https://github.com/grafana/grafana/pull/96781/files#diff-401f8230f984d481214007ecab2a67d1687bb3accde24fb140c027720b04c1e5R183

File: public/app/features/transformers/docs/content.ts [hunk 3]
```diff
@@ -179,6 +179,9 @@ In the field mapping specify:
 | Color | Value mappings / Color | All values |
 
 Grafana builds value mappings from your query result and applies them to the real data query results. You should see values being mapped and colored according to the config query results.
+
+> **Note:** When using this transformation for thresholds, the visualization will continue to use the panel's base threshold.
```
Comment (gelicia) [inline public/app/features/transformers/docs/content.ts:183]: @imatwawana any thoughts on this?

Comment (dprokop) [reply public/app/features/transformers/docs/content.ts:183]: Looks good @gelicia !

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 4]
```diff
@@ -1216,6 +1218,8 @@ Select this option to transform the time series data frame from the long format
 | 2023-01-01 00:00:00 | 10     | 20     |
 | 2023-01-01 01:00:00 | 15     | 25     |
 
+> **Note:** This transformation is available in Grafana 7.5.10+ and Grafana 8.0.6+.
```
Comment (gelicia) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:1221]: @imatwawana I followed the instructions to generate this file and a couple unrelated lines popped in, is this okay? Did I do something wrong?

Comment (leeoniya) [reply docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:1221]: i had this before and just manually reverted those changes
